File: taurus-protect-sdk-typescript/src/models/air-gap.ts
/**
 * Air gap models for Taurus-PROTECT SDK.
 *
 * These models represent the data structures used for air-gapped signing
 * workflows with cold HSM (Hardware Security Module) devices.
 */

/**
 * Options for exporting requests for cold HSM signing.
 */
export interface GetOutgoingAirGapOptions {
  /**
   * List of request IDs to export for cold HSM signing.
   * These requests must be in HSM-ready state.
   */
  requestIds: string[];

  /**
   * Optional ECDSA signature of the request hashes.
   * Format: base64(ecdsa_sign(sha256([hex(sha256(req1)),hex(sha256(req2)),...]))
   */
  signature?: string;
}

/**
 * Options for exporting addresses for cold HSM signing.
 */
export interface GetOutgoingAirGapAddressOptions {
  /**
   * List of address IDs to export for cold HSM signing.
   */
  addressIds: string[];
}

/**
 * Options for submitting signed responses from the cold HSM.
 */
export interface SubmitIncomingAirGapOptions {
  /**
   * The signed payload from the cold HSM (base64-encoded).
   * This is the IncomingAirGapFile_SignedPayload from the HSM.
   */
  payload: string;

  /**
   * Optional signature of the air-gap importer (base64-encoded).
   */
  signature?: string;
}

File: taurus-protect-sdk-go/pkg/protect/model/airgap.go
package model

// GetOutgoingAirGapRequest contains parameters for exporting HSM ready requests for cold HSM.
type GetOutgoingAirGapRequest struct {
	// RequestIDs is the list of request IDs that have been approved to be signed by the cold HSM.
	RequestIDs []string `json:"request_ids,omitempty"`
	// RequestSignature is the signature of the requests (optional).
	// Format: base64(ecdsa_sign(sha256([hex(sha256(req1_metadata)),hex(sha256(req2_metadata)),...hex(sha256(reqN_metadata))])))
	RequestSignature string `json:"request_signature,omitempty"`
	// AddressIDs is the list of address IDs to be signed by the cold HSM (optional).
	AddressIDs []string `json:"address_ids,omitempty"`
}

// GetOutgoingAirGapResult contains the result of exporting HSM ready requests.
type GetOutgoingAirGapResult struct {
	// Data contains the binary payload to be transmitted to the cold HSM.
	Data []byte `json:"data"`
}

// SubmitIncomingAirGapRequest contains parameters for importing signed requests from cold HSM.
type SubmitIncomingAirGapRequest struct {
	// Payload is the base64-encoded signed payload from the air-gap importer.
	Payload string `json:"payload"`
	// Signature is the base64-encoded signature of the air-gap importer.
	Signature string `json:"signature"`
}
